### ISSUE TITLE
Using python 3.7 fails on a Mac

### ISSUE THREAD

**Issue by DonJayamanne** [davidhalter/jedi#1257]
When using Python3.7 on a Mac, we (Python Extension for VS Code) throws the following error:

```
stderr jediProxy, Error (stderr)     _assert_safe(path, safe)
  File "/Users/donjayamanne/.vscode-insiders/extensions/pythonVSCode/pythonFiles/jedi/api/environment.py", line 352, in _assert_safe

stderr jediProxy, Error (stderr)     "The python binary is potentially unsafe.")
jedi.api.environment.InvalidPythonEnvironment: The python binary is potentially unsafe.
```

**Current solution:**
Change instantiation of Jedi from
* `jedi.api.environment.create_environment(sys.executable)` to 
* `jedi.api.environment.Environment(sys.executable)` 

Could be related to https://github.com/davidhalter/jedi/issues/1177

**Comment by davidhalter** [comment]
Can you try he typeshed branch? I might enable travis testing for mac if this is really still an issue.

**Comment by Dovermore** [comment]
I have the same issue. Using anaconda distribution of jedi.

**Comment by Dovermore** [comment]
I find the cause, trying to fix it now.

**Comment by Dovermore** [comment]
On my machine with conda distribution of python (both 3.6 and 3.7) will throw similar error. The reason being the TODO for `get_system_environment` won't pick up conda installed python.

Could it be the same problem here? I've done the fix for the issue on my machine. I will create a pull request for this.

It's my first time contributing. So I'm not familiar with the process of contributing. Please inform me if you can.

I haven't added the test yet. I'm not too sure how to do this. It will be great if you can point me to something I can get started with.

**Comment by davidhalter** [comment]
Can you guys try with latest master? Quite a lot has changed (stubs were merged).

@Dovermore You can just create a Pull Request. No worries, we can then discuss how to proceed. Don't worry about tests, we can worry about that later as well.

**Comment by davidhalter** [comment]
@Dovermore I see what you're trying to change in #1324. However, I fail to see what your exact issue is. What's the error messages you are seeing?

Why are the other Python executables on path not working?

**Comment by Dovermore** [comment]
@davidhalter Hi, on my system it's only checking for executables in 
/bin/, /local/bin/ 
and don't check for executables in
/anaconda/bin/
or some other activated environments (by conda).

The error message is the same.
`stderr jediProxy, Error (stderr)     "The python binary is potentially unsafe.")
jedi.api.environment.InvalidPythonEnvironment: The python binary is potentially unsafe.`

**Comment by davidhalter** [comment]
Which call are you using? `get_system_environment()`?

**Comment by Dovermore** [comment]
I'm not too sure about what you meant by which call I'm using. 

The version of jedi I installed (should be the newest release) before editing is using the default `find_system_environments()` to find valid python envs. It then calls `get_system_environment(version)` for all python versions.
It fails here. Because I think the `which` command from `jedi._compatibility` only detects the first python it can find.
So the overall results for *nix system become:
1. check with the defined safe path.
2. check the first python in path.

Thus I added the support for flag `a` for `which`.

**Comment by davidhalter** [comment]
> Because I think the which command from jedi._compatibility only detects the first python it can find.

That's expected and I currently don't feel like we should change that. It's just a convenience function. There's two different ways you can achieve what you want:

1. Use `create_environment` to create an environment for a specific path.
2. Fix your Python installation so that the anaconda one is found before the other one (or there's just one).

**Comment by davidhalter** [comment]
@DonJayamanne Please try again on master.

**Comment by Dovermore** [comment]
Hi, Thanks for the reply! I will try that.

**Comment by davidhalter** [comment]
I feel liket his is fixed, 3.7 support is here with 0.14.0 and if it doesn't work on mac, please let me know.
